docs/guidelines.md

```diff
@@ -0,0 +1,195 @@
+
+### 1.3. Ownership
+
+Each team should be responsible for maintaining the workflows or actions they contribute to this repository.
+When contributing workflows or actions, add your team as a [code owner](../.github/CODEOWNERS), unless agreed upon. This
+helps ensure appropriate review coverage and accountability.
+
+## 2. Security
+
+Please see [security guidelines](security.md).
+
+## 3. Style and formatting
+
+All workflows and actions must follow consistent formatting rules to improve readability and maintainability.
+Consistency also helps reduce unnecessary diff noise during reviews.
+
+- Format YAML files using [`prettier`](https://prettier.io/) with the default config.
+- Use the `.yml` extension for YAML files (not `.yaml`).
+- Indent using 2 spaces consistently.
+- Add inline comments where logic may not be obvious.
+
+## 4. Workflows
+
+- Repository-specific workflows: `.github/workflows/`
+- Workflow templates for reuse: `.github/workflow-templates/`
+- Reusable workflows: `hemilabs/actions/.github/workflows/`
+
+Refer to [Reusable actions and workflows](#1-reusable-actions-and-workflows) for reusable workflows.
+
+### 4.1. File name
+
+Use lowercase kebab-case file names with the `.yml` extension. Keep names short but descriptive.
+
+The file name should roughly match the workflow name.
+
+**Examples:**
+
+- `go.yml`: Build, lint and test a Go project.
+- `js.yml`: Build, lint and test a JavaScript project.
+- `npm-publish.yml`: Publish an NPM project.
+- `kustomize-deploy.yml`: Deploy using Kustomize.
+
+### 4.2. Workflow Name
+
+Workflow names should be a short, clear label - usually a noun (e.g. "Go" or "JavaScript") - that describes the type of
+project or task the workflow is for.
+
+```yaml
+name: "Go"
+name: "JavaScript"
+name: "Deploy"
+```
+
+Workflow names appear with the job name in the GitHub UI as `<workflow> / <job>`, so both should be readable and
+descriptive together.
+
+**Good examples:**
+
+- `Go / Build`
+- `JavaScript / Lint`
+- `Deploy / Staging`
+
+### 4.3. Jobs
+
+Job names should normally be verbs to describe the action being taken.
+
+**Examples:**
+
+- `Build` - Builds the project.
+- `Test` - Runs tests.
+- `Deploy` - Deploys to an environment.
+- `Lint` - Runs linters.
+- `Publish` - Publish a package.
+
+For context-specific jobs (e.g. when using matrix), consider using parentheses:
+
+- `Build (${{ env.GO_VERSION }})` -> `Go / Build (1.24.2)`
+- `Deploy (staging)`
+
+### 4.5. Steps
+
+Jobs consist of steps, each of which should be:
+
+- Clear in intent.
+- Named descriptively.
+- Ordered logically.
+- Commented if they contain non-obvious logic.
+
+It is good practice to use step names that clearly describe their function, improving maintainability and readability.
+
+Step keys should be ordered as:
+
+- `name`
+- `id`
+- `if`
+- `shell`
+- `working-directory`
+- `env`
+- `uses` / `run`
+- `with`
+
+Ordering keys consistently improves readability and reduces unnecessary diffs during reviews.
```

Comment:
The workflows the JS team authors and maintains have their keys sorted in the same order of the Actions docs:

https://docs.github.com/en/actions/writing-workflows/workflow-syntax-for-github-actions
https://docs.github.com/en/actions/sharing-automations/creating-actions/metadata-syntax-for-github-actions

References to both docs should be added here.